src/mmw/js/src/core/models.js

```diff
@@ -238,6 +239,77 @@ var LayerModel = Backbone.Model.extend({
     }
 });
 
+var WeatherStationLayerModel = LayerModel.extend({
+    defaults: {
+        bringToFront: true, // Ensures popups work when station is within project AOI.
+        activeWeatherStations: null,
+        activeWeatherStationPoints: null
+    },
+
+    // Sets active weather stations. Used to change the visual appearance
+    // of any stations participating as modeling input, and to set the
+    // distance of the weather station from an AOI into the popup.
+    setActiveWeatherStations: function(weatherStations) {
+        var leafletLayer = this.get('leafletLayer'),
+            activeWeatherStations = {};
+
+        if(this.activeWeatherStations) {
+            this.clearActiveWeatherStations();
+        }
+
+        _.forEach(weatherStations, function(ws) {
+            activeWeatherStations[ws.station] = ws;
+        });
+
+
+        this.set('activeWeatherStations', activeWeatherStations);
+
+        if(leafletLayer) {
+            var activeWeatherStationPoints =
+                weatherStationLayer.Layer.setActiveWeatherStations(leafletLayer, activeWeatherStations);
+            this.set('activeWeatherStationPoints', activeWeatherStationPoints);
+        }
+    },
+
+    clearActiveWeatherStations: function() {
+        var leafletLayer = this.get('leafletLayer'),
+            prevActiveWeatherStations = this.get('activeWeatherStations');
+
+        this.set('activeWeatherStations', null);
+        this.set('activeWeatherStationPoints', null);
+
+        if(leafletLayer && prevActiveWeatherStations) {
```

Comment:
New change - guard against the case where this method is called when `this.get('activeWeatherStations')` is null.